key-share/src/lib.rs

```diff
@@ -441,6 +441,33 @@ impl<E: Curve> DirtyCoreKeyShare<E> {
     }
 }
 
+#[cfg(feature = "hd-wallets")]
+impl<E: Curve> DirtyCoreKeyShare<E> {
+    /// Checks whether the key is HD-capable
+    pub fn is_hd_wallet(&self) -> bool {
+        (**self).is_hd_wallet()
+    }
```

Comment:
Aren't these methods available automatically with DirtyCoreKeyShare implementing `AsRef<KeyInfo>`?

Reply:
They should be, but it’s still a breaking change to remove a method in public API (even if it’s still available via derefing)

Reply:
E.g. using DirtyCoreKeyShare::is_hd_wallet compiles before the removal, but doesn’t compile after the removal, derefing doesn’t work here

Reply:
Since when do you care about breaking the API? :>

Reply:
Since always! I never was a fan of updating all the upstream deps, I only do that when there's no choice